packages/engine/src/networking/functions/WorldNetworkActionReceptor.test.ts

```diff
@@ -184,142 +185,71 @@ describe('WorldNetworkActionReceptors', () => {
-
-  //     const hostIndex = 0
-
-  //     network.peers.set(Engine.instance.worldNetwork.hostId, {
-  //       userId: Engine.instance.worldNetwork.hostId,
-  //       index: hostIndex
-  //     })
-
-  //     const objNetId = 3 as NetworkId
-  //     const objPrefab = 'generic prefab'
-
-  //     const action = WorldNetworkAction.spawnObject({
-  //       $from: Engine.instance.worldNetwork.hostId, // from host
-  //       prefab: objPrefab, // generic prefab
-  //       networkId: objNetId
-  //     })
-
-  //     WorldNetworkActionReceptor.receiveSpawnObject(action)
-  //     createAvatar(action)
-
-  //     const networkObjectQuery = defineQuery([NetworkObjectComponent])
-  //     const networkObjectOwnedQuery = defineQuery([NetworkObjectOwnedTag])
-
-  //     const networkObjectEntities = networkObjectQuery(world)
-  //     const networkObjectOwnedEntities = networkObjectOwnedQuery(world)
-
-  //     assert.equal(networkObjectEntities.length, 1)
-  //     assert.equal(networkObjectOwnedEntities.length, 0)
-  //     assert.equal(
-  //       getComponent(networkObjectEntities[0], NetworkObjectComponent).authorityPeerID,
-  //       Engine.instance.worldNetwork.hostId
-  //     )
-
-  //     WorldNetworkActionReceptor.receiveRequestAuthorityOverObject(
-  //       WorldNetworkAction.requestAuthorityOverObject({
-  //         $from: userId, // from user
-  //         object: {
-  //           ownerId: Engine.instance.worldNetwork.hostId,
-  //           networkId: objNetId
-  //         },
-  //         requester: userId,
-  //         $topic: NetworkTopics.world
-  //       }),
-  //       world
-  //     )
-
-  //     ActionFunctions.clearOutgoingActions(network.topic)
-  //     ActionFunctions.applyIncomingActions()
-  //     executeSystems(0)
-
-  //     assert.equal(networkObjectEntities.length, 1)
-  //     assert.equal(networkObjectOwnedEntities.length, 0)
-
-  //     assert.equal(getComponent(networkObjectEntities[0], NetworkObjectComponent).networkId, objNetId)
-  //     assert.equal(
-  //       getComponent(networkObjectEntities[0], NetworkObjectComponent).authorityPeerID,
-  //       Engine.instance.worldNetwork.hostId
-  //     )
-  //     assert.equal(hasComponent(networkObjectEntities[0], NetworkObjectOwnedTag), false)
-  //   })
-  // })
+      const networkObjectEntitiesBefore = networkObjectQuery()
+      const networkObjectOwnedEntitiesBefore = networkObjectOwnedQuery()
+
+      assert.equal(networkObjectEntitiesBefore.length, 1)
+      assert.equal(networkObjectOwnedEntitiesBefore.length, 0)
+
+      assert.equal(getComponent(networkObjectEntitiesBefore[0], NetworkObjectComponent).ownerId, hostUserId)
+      assert.equal(getComponent(networkObjectEntitiesBefore[0], NetworkObjectComponent).authorityPeerID, peerID)
+
+      WorldNetworkActionReceptor.receiveRequestAuthorityOverObject(
+        WorldNetworkAction.requestAuthorityOverObject({
+          $from: hostUserId, // from host
```

Comment:
we should be passing in `userId` here

Reply:
I think passing `userId` will not do the transfer because the `receiveTransferAuthorityOfObject()` function is checking if action's [`$from === ownerId`](https://github.com/EtherealEngine/etherealengine/blob/dev/packages/engine/src/networking/functions/WorldNetworkActionReceptor.ts#L135)